src/py42/clients/securitydata.py

```diff
@@ -278,13 +213,83 @@ def stream_file_by_md5(self, checksum):
         """
         response = self._search_by_hash(checksum, MD5)
         events = response[u"fileEvents"]
-        if not len(events):
+        info = _get_version_lookup_info(events)
+        if not len(events) or not info:
             raise Py42ChecksumNotFoundError(response, u"MD5", checksum)
-        sha256_hash = events[0][u"sha256Checksum"]
-        return self._stream_file(
-            self._find_file_versions(checksum, sha256_hash), checksum
+        return self._stream_file(checksum, info)
+
+    def _search_by_hash(self, hash, type):
+        query = FileEventQuery.all(type.eq(hash))
+        query.sort_key = "eventTimestamp"
+        query.sort_direction = "desc"
+        response = self.search_file_events(query)
+        return response
+
+    def _stream_file(self, checksum, version_info):
```

Comment:
This is the fix to the stream by hash logic that I mentioned. Before, we were writing a generator to loop through a list of versions and try each one. It turns out that the service accepts a list of versions and does that for you, so because of that we were able to get rid of a lot of extra log + retry logic.